packages/react-dom/src/events/plugins/ChangeEventPlugin.js

```diff
@@ -280,8 +284,16 @@ function extractEvents(
 ) {
   const targetNode = targetInst ? getNodeFromInstance(targetInst) : window;
 
+  let isCustomComponentTag = false;
+  if (enableCustomElementPropertySupport && targetInst) {
+    isCustomComponentTag = isCustomComponent(
+      targetInst.elementType,
+      targetInst.pendingProps,
+    );
+  }
+
   let getTargetInstFunc, handleEventFunc;
-  if (shouldUseChangeEvent(targetNode)) {
+  if (shouldUseChangeEvent(targetNode) || isCustomComponentTag) {
```

Comment:
Maybe let’s move this logic to shouldUseChangeEvent and pass it there?

Reply:
done